lib/fs.js

```diff
@@ -62,13 +74,8 @@ const DEBUG = process.env.NODE_DEBUG && /fs/.test(process.env.NODE_DEBUG);
 const errnoException = util._errnoException;
 
 function getOptions(options, defaultOptions) {
-  if (options === null || options === undefined ||
-      typeof options === 'function') {
-    return defaultOptions;
-  }
-
   if (typeof options === 'string') {
-    defaultOptions = util._extend({}, defaultOptions);
+    defaultOptions = shallowClone(defaultOptions);
```

Comment:
I haven't tested `Object.assign()` with V8 5.7 but I would guess it's still slow. `shallowClone` is really fast because it literally copies everything as-is (including frozen/sealed state, property attributes, etc.). From my own separate benchmarks, `shallowClone()` was something like at least twice as fast as `for (... in ...)` and `util._extend()` uses `Object.keys()` to iterate instead which the last time I tested is even slower than `for (... in ...)` in many/most cases.